modules/gapi/src/backends/ie/giebackend/gieapi.hpp

```diff
@@ -0,0 +1,159 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2020 Intel Corporation
+
+#ifndef OPENCV_GAPI_IEAPI_HPP
+#define OPENCV_GAPI_IEAPI_HPP
+
+#ifdef HAVE_INF_ENGINE
+
+#include <inference_engine.hpp>
+
+#include <opencv2/core/utility.hpp>
+#include <opencv2/core/utils/logger.hpp>
+
+#include <ade/util/range.hpp>
+#include <ade/util/zip_range.hpp>
+
+#include "opencv2/gapi/infer/ie.hpp"
+
+namespace IE = InferenceEngine;
+
+namespace cv {
+namespace gapi {
+namespace ie {
+namespace wrap {
+
+std::vector<std::string> getExtensions(const cv::gapi::ie::detail::ParamDesc& params);
+
+#if INF_ENGINE_RELEASE < 2020000000  // < 2020.1
+// Load extensions (taken from DNN module)
+std::vector<std::string> getExtensions(const cv::gapi::ie::detail::ParamDesc& params) {
```

Comment:
in order to avoid the "multiple definitions" problem at the linker level, all these functions needs to be either
* `static` (old-schoolish, but correct)
* OR `inline` (not quite correct)
* OR wrapped into an anonymous namespace (the best way to do assuming you want to keep the implementations here)
* OR moved to a new `.cpp` file which gets compiled only one. Probably the most right thing to do overall.

Reply:
inline functions are kept as is and others are moved to .cpp